src/main/java/programminglife/model/Node.java

```diff
@@ -0,0 +1,70 @@
+package programminglife.model;
+
+import java.util.HashSet;
+import java.util.Scanner;
+import java.util.Set;
+
+/**
+ * Created by marti_000 on 25-4-2017.
+ */
+public class Node {
+    private int id;
+    private String sequence;
+
+    private Set<Node> parents;
+    private Set<Node> children;
+
+    public Node(int id) {
+        this(id, new String(), new HashSet<>(), new HashSet<>());
```

Comment:
new String() allocates a new String. Please use "" as that will reuse the internal buffer for the empty string.